### PR TITLE
docs: establish Artifacts WG

### PR DESCRIPTION
**Pull request by afflom**
Establish the Artifacts WG under CNCF TAG App Delivery.

Closes: #368

### CODE REVIEW

**Comment by afflom** [comment]
@joshgav ptal

**Comment by joshgav** [comment]
@afflom thank you for preparing this charter and WG proposal! :tada: The charter you proposed includes the proposed purpose, goals, and activities of the WG :+1:, and we included initial problem statements/user scenarios as an appendix too to inform reviewers.

Next we'll:

1. Ask TAG App Delivery leaders and [TOC](https://github.com/cncf/toc) members to review and comment on this PR
2. Merge this PR and officially start the WG when we have consensus
3. Select WG leaders, meeting times and other logistics; though we already have folks in mind to nominate.
4. We'll continue the meetings and initial work we've already started.

Let's note here that the content of the charter is based on text and discussions in comments from <https://docs.google.com/document/d/1w_lo2RZDKeEzQg4DMV-9Tq4ir_znONj_ypJ27CUfMgY/>. Folks may want to review those comments to get full background.

As the WG briefly also discussed, we didn't include the list of interested parties from the draft doc so as not to permanently clutter _this_ repo and doc. However, we should think about how to notify and invite them to our meetings and contribute to our work. Perhaps we can CC them here and invite them all to the <https://lists.cncf.io/g/cncf-tag-app-delivery/topics> listserv and [#wg-artifacts Slack channel](https://cloud-native.slack.com/archives/C04UQDWS4M7).

**Comment by afflom** [comment]
@joshgav signed-off

**Review by joshgav — APPROVED** [review]

**Review by scottrigby — APPROVED** [review]
Looks great! And thanks for following up after the kccnc meetings. Tl;dr, 🚢 👍 

One note: this md doc doesn't include the "existing work" section from the [Charter draft](https://docs.google.com/document/d/1w_lo2RZDKeEzQg4DMV-9Tq4ir_znONj_ypJ27CUfMgY/edit?usp=sharing), but I think that's fine we can keep a link to that and can fold this info into one of the work products produced by the group.

**Comment by scottrigby** [comment]
> As the WG briefly also discussed, we didn't include the list of interested parties from the draft doc so as not to permanently clutter _this_ repo and doc. However, we should think about how to notify and invite them to our meetings and contribute to our work. Perhaps we can CC them here and invite them all to the https://lists.cncf.io/g/cncf-tag-app-delivery/topics listserv and [#wg-artifacts Slack channel](https://cloud-native.slack.com/archives/C04UQDWS4M7).

@joshgav We could revisit including an interested parties doc for each WG in this repo? (Related issue https://github.com/cncf/tag-app-delivery/pull/256#issuecomment-1462693018). Or perhaps it's worth adding a "Working Groups" section to the CNCF Community Groups site so that folks can sign up directly for activity by specific WGs. Either way, something we can follow-up on after this WG is fully established 😸

File: artifacts-wg/charter/charter.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# Alignment with TAG App Delivery and CNCF
+TAG App Delivery's charter [2] calls on the TAG to develop best practices, foster collaboration between projects, improve interoperability and propose new initiatives and projects related to the following domains which this working group would target.
+
+* Application bundling and deployment
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on official specifications in the Open Containers Initiative (OCI) itself.
```
Comment (SteveLasker) [inline artifacts-wg/charter/charter.md:23]: nit: official is redundant

File: artifacts-wg/charter/charter.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/artifacts
+* Develop a pattern for dynamically discovering existing Artifact schemas
+This in contrast to asking all schema developers to publish media types via IANA.
```
Comment (SteveLasker) [inline artifacts-wg/charter/charter.md:28]: Iana registration feels like a specific thing that might be part of the discussion, but is it a goal to remove?
Iana registration was used to solve a clearing house of ownership for a specific `artifactType`, It solves the conflicting file extensions type problem. What keeps someone from defining different versions of "helm", or "spdx"? 
See [Defining a Unique Artifact Type](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md#defining-a-unique-artifact-type)
To avoid bike shedding specifics here, perhaps we can remove line 28, and explore the dynamic discovery?

**Review by SteveLasker — COMMENTED** [review]
Thanks @afflom for getting this started. 
A few minor nits to generalize the charter.

File: artifacts-wg/charter/charter.md (hunk 1 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md:23]: Agreed

File: artifacts-wg/charter/charter.md (hunk 2 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md:28]: @SteveLasker, I agree that we should not call out IANA here, however I do think that defining an extensibility model is important for the working group to have as a goal. Ideally, it would take administrative burden off of registry operators and publishers by dynamically registering schemas as they are used to describe artifacts and their relationships.

File: artifacts-wg/charter/charter.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* Application bundling and deployment
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on official specifications in the Open Containers Initiative (OCI) itself.
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/artifacts
```
Comment (sudo-bmitch) [inline artifacts-wg/charter/charter.md:26]: I'm expecting this repo to be archived in the not too distant future:
https://github.com/opencontainers/artifacts/issues/61
https://github.com/opencontainers/artifacts/issues/63

We may want to link to image-spec instead.

Comment (SteveLasker) [reply artifacts-wg/charter/charter.md:26]: Having this group demonstrates theres been a lot of value in OCI Artifacts, so it's not clear why we would archive the repo.

File: artifacts-wg/charter/charter.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+* Develop a pattern for dynamically discovering existing Artifact schemas
+This in contrast to asking all schema developers to publish media types via IANA.
+* Contribute to a common data model for Artifacts in OCI
```
Comment (sudo-bmitch) [inline artifacts-wg/charter/charter.md:29]: There's going to be more than one data model for artifact content. This implies they will all have a single model.

File: artifacts-wg/charter/charter.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on official specifications in the Open Containers Initiative (OCI) itself.
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/artifacts
+* Develop a pattern for dynamically discovering existing Artifact schemas
```
Comment (sudo-bmitch) [inline artifacts-wg/charter/charter.md:27]: There's dynamically discovering schemas, and dynamically discovering the artifacts themselves. The former may be both more difficult and less useful to end users.

File: artifacts-wg/charter/charter.md (hunk 4 above)
Comment (sudo-bmitch) [reply artifacts-wg/charter/charter.md:29]: If this is getting into defining a new OCI manifest type, OCI is already planning to look at a replacement for the artifact manifest after the image-spec 1.1 release. It's probably better to do that over in OCI.

File: artifacts-wg/charter/charter.md (hunk 2 above)
Comment (ktarplee) [reply artifacts-wg/charter/charter.md:28]: In an effort to not over promise we might want to make a section called "Stretch Goals" to cover topics including decentralization and federation.  We could also include as a stretch goal something about how OCI could be a package indices replacement for PyPI, Apt, etc.

**Review by sajayantony — APPROVED** [review]

File: artifacts-wg/charter/charter.md (hunk 4 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md:27]: I think that's the reason that this working group was originally pitched to the OCI maintainers. It would be unfortunate if the next OCI data model was not compatible with the search API. With consideration for that prior attempt at establishing this working group under OCI, do you have any suggestions for combining these efforts or at least ensuring this working group's output is a consideration in the development of the next OCI data model?

File: artifacts-wg/charter/charter.md (hunk 5 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md]: I have thoughts on this, but since the line is removed, I'll save them for when we discuss implementation.

File: artifacts-wg/charter/charter.md (hunk 4 above)
Comment (sudo-bmitch) [reply artifacts-wg/charter/charter.md:27]: Our focus right now is on the 1.1 release. Once that's done, merging efforts would be great.

File: artifacts-wg/charter/charter.md (hunk 3 above)
Comment (sudo-bmitch) [reply artifacts-wg/charter/charter.md]: I think that discussion is best had in the linked issues. The short version is that image-spec has added a lot of details on how to package content other than images, and it has made changes that aren't included in the artifact repo. Rather than trying to keep the two in sync, I think it makes sense to consolidate to one repo.

File: artifacts-wg/charter/charter.md (hunk 2 above)
Comment (ktarplee) [reply artifacts-wg/charter/charter.md]: I think the "package index" idea is covered under the "Search Query API" goal.  It might make sense to add "package index" to the "Activities" section where we talk about npm, PyPI, etc.  We only discuss the artifacts in that activity (i.e., how we might store them) and not how we might search for them (i.e., the package index functionality).

File: artifacts-wg/charter/charter.md (hunk 4 above)
Comment (sudo-bmitch) [reply artifacts-wg/charter/charter.md:27]: This could be rephrased a bit to focus on the need to package various types of artifacts vs defining a single model (depending on the level where we implement this, we may still have a single model, we may leverage an existing model, or there may be multiple models).

```suggestion
* Define data models for storing Artifacts in OCI
```

File: artifacts-wg/charter/charter.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on specifications in the Open Containers Initiative (OCI) itself.
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/image-spec
+* Contribute to a common data model for Artifacts in OCI
+* Seek to minimize or avoid any specification changes
```
Comment (sudo-bmitch) [inline artifacts-wg/charter/charter.md:28]: Consider removing since this covers how rather than why. We may still do this, but it can also be an arbitrary constraint on our work.

File: artifacts-wg/charter/charter.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+* Define an Artifact Query API based on specific or common data models as an OCI distribution-spec extension
```
Comment (sudo-bmitch) [inline artifacts-wg/charter/charter.md:29]: I'm trying to figure out how to rephrase this towards the problem being solved vs the how we are solving it. I'm not excited about my phrasing, but I want to get to a search/query API that may need more parameters than exist in current OCI query APIs.

```suggestion
* Define an API to query for Artifacts using parameters needed for the different types of Artifact tooling.
```

**Review by sudo-bmitch — COMMENTED** [review]
Some suggestions based on Justin's feedback.

File: artifacts-wg/charter/charter.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully learned and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development by users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
+
+To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects offering packaging, distribution and deployment mechanisms for bundles of configuration and code.
+
+**The WG will:**  
+* **Develop common patterns**
```
Comment (AlexsJones) [inline artifacts-wg/charter/charter.md:9]: Will you really be developing patterns rather than observing and notating what is in use?

File: artifacts-wg/charter/charter.md (hunk 4 above)
Comment (AlexsJones) [reply artifacts-wg/charter/charter.md:27]: I would caution activities that define a deliverable outcome as that's what your success will be hinged upon.
Defining OCI for artefacts is going to be a long process of getting people to agree with each other. 
Perhaps you can save yourself with wording here
```suggestion
Identify an exemplar data model for storing artefacts in OCI
```

File: artifacts-wg/charter/charter.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/image-spec
+* Contribute to a common data model for Artifacts in OCI
+* Seek to minimize or avoid any specification changes
+* Define an Artifact Query API based on specific or common data models as an OCI distribution-spec extension
+* Publish a prototypical implementation of the Artifact Query API service within a registry and a prototypical implementation of a client for that API service
+
+# Activities
+The working group intends to carry out (but is not limited to) the following:
+* Gather stakeholders using OCI specifications distribution-spec and image-spec to package and distribute artifacts to document their bundle layouts and content schemas and seek common patterns.
+* Why? Reduce cognitive load on end users, reduce errors in configuration, and optimize ongoing development in this space.
```
Comment (AlexsJones) [inline artifacts-wg/charter/charter.md:35]: This isn't an activity is it?

Comment (afflom) [reply artifacts-wg/charter/charter.md:35]: Good catch. I've removed it.

File: artifacts-wg/charter/charter.md (hunk 3 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md]: I've removed the reference altogether in my re-imagining of the goals section.

File: artifacts-wg/charter/charter.md (hunk 4 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md]: @sudo-bmitch @AlexsJones, based on your suggestions and the feedback from @justincormack, I've tried to better address this in the revised goals.

File: artifacts-wg/charter/charter.md (hunk 6 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md]: Removed

File: artifacts-wg/charter/charter.md (hunk 7 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md]: I used this as inspo in my goals refactor. Can you PTAL?

File: artifacts-wg/charter/charter.md (hunk 8 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md:9]: I'm hoping that my refactor on the goals section clarifies this a bit. Can you PTAL and let me know if you'd like this bit about patterns reworked?

**Comment by afflom** [comment]
I've reworked the goals based on all of the feedback here and in the Artifacts WG meetings. In this rework, I've distilled the goals into 3 "stories". Those are research, design, and prototype. Please let me know if these work or if I should revert to the previous goals format.

File: artifacts-wg/charter/appendix.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Artifacts WG Charter Appendix 
+
+## Problem Statements
+* As an artifact consumer I need to discover attributes about every component used or potentially used in my system. I need consistent practices and patterns for querying for these attributes across many artifact types.
+* As an artifact provider I need to deliver complete components to a runtime platform which may be completely disconnected from the environment where those components are developed. I want consistent practices for bundling and deploying different component types.
```
Comment (sabre1041) [inline artifacts-wg/charter/appendix.md:5]:
```suggestion
* As an artifact provider, I need to deliver a complete set of components to a runtime platform which may be fully disconnected from the environment from where those components are developed or assembled. I want consistent practices for bundling and deploying different component types.
```

File: artifacts-wg/charter/appendix.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+* As a security analyst I need to understand and verify the content of bundles used in software and deployed on my platforms.
```
Comment (sabre1041) [inline artifacts-wg/charter/appendix.md:6]:
```suggestion
* As a security analyst, I need to understand and verify the content of bundles used in software and deployed on my platforms.
```

File: artifacts-wg/charter/appendix.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
...
+* [Proposed] As a platform operator I need a consistent pattern and implementation for deploying bundles to provide additional platform capabilities. For example, I need a consistent pattern for adding extension controller-managers (operators) to Kubernetes.
```
Comment (sabre1041) [inline artifacts-wg/charter/appendix.md:7]:
```suggestion
* [Proposed] As a platform operator, I need a consistent pattern and implementation approach for deploying bundles to provide additional platform capabilities. For example, I need a defined pattern for adding extension controller-managers (operators) to Kubernetes.
```

File: artifacts-wg/charter/charter.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
+# Purpose
+In the distributed world of cloud-native computing different artifacts and packages are used to transport configuration and code for the many services and capabilities that comprise and support workloads and applications. As an example, today ArtifactHub advertises that it can crawl about 20 such artifact types [1]. These "cloud-native" bundles add to the previous proliferation of package and artifact types for software dependencies such as Maven and npm and for system packages like deb and rpm.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:2]:
```suggestion
In the distributed world of cloud-native computing, different artifacts and packages are used to transport configuration and code for the many services and capabilities that comprise and support various types of workloads. As an example, today ArtifactHub advertises that it can crawl about 20 such artifact types [1]. These "cloud-native" bundles adds to the previous proliferation of package and artifact types for software dependencies, such as Maven and npm and for system packages, like deb and rpm.
```

File: artifacts-wg/charter/charter.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully learned and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development by users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:4]: Are we attempting to limit scope to "cloud native"? I understand that this WG is within the CNCF realm, but for components in the application packaging space, like Maven and npm, there is no assumption that they will be delivered to cloud native targets.

Comment (sabre1041) [inline artifacts-wg/charter/charter.md:4]:
```suggestion
This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully tailored and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development for users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
```

File: artifacts-wg/charter/charter.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects offering packaging, distribution and deployment mechanisms for bundles of configuration and code.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:6]:
```suggestion
To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects to offer packaging, distribution and deployment mechanisms for bundles of configuration manifests and source code.
```

File: artifacts-wg/charter/charter.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+**The WG will:**
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:8]: Should there also be any discussion related to advocating for the approaches that come out of the working group with others in the CNCF community?

File: artifacts-wg/charter/charter.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+
+This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully learned and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development by users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
+
+To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects offering packaging, distribution and deployment mechanisms for bundles of configuration and code.
+
+**The WG will:**  
+* **Develop common patterns**  
+*  **Advocate for innovative projects**  
+*  **Gather end user feedback**
+**for packaging, discovery, distribution, and deployment mechanisms for** **cloud-native software artifacts.**
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:12]:
```suggestion
**For packaging, discovery, distribution, and deployment mechanisms for** **cloud-native software artifacts.**
```

Comment (sabre1041) [reply artifacts-wg/charter/charter.md:12]: What is this goal indicating?

File: artifacts-wg/charter/charter.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+# Alignment with TAG App Delivery and CNCF
+TAG App Delivery's charter [2] calls on the TAG to develop best practices, foster collaboration between projects, improve interoperability and propose new initiatives and projects related to the following domains which this working group would target.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:15]:
```suggestion
TAG App Delivery's charter [2] calls on the TAG to develop best practices, foster collaboration between projects, improve interoperability, and propose new initiatives and projects related to the following domains for which this working group would target.
```

File: artifacts-wg/charter/charter.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+
+* Application bundling and deployment
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on specifications in the Open Containers Initiative (OCI) itself.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:23]: Recommend removing the explicit mention of OCI as it is a bit too implementation specific. While yes, OCI is certainly an initial target, should it be in the charter for the WG?

File: artifacts-wg/charter/charter.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+
+# Goals
+* Research - Collect existing software packaging schemas, use-cases, and other evidence to aid in the design of a data model and search API for artifact management. 
+* Design - Define a conceptual model and search API from the lessons learned from research, that will be used to build prototypes and inform future specifications.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:27]:
```suggestion
* Design - Define a conceptual model and search API from the lessons learned from the research phase, that will be used to build prototypes and inform future specifications.
```

File: artifacts-wg/charter/charter.md [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+* Prototype - Develop prototype implementations of a data model and search API based on the work from the design phase to validate the design and to provide a reference implementation for future specifications.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:28]:
```suggestion
* Prototype - Develop prototype implementations of a data model and search API based on the work from the design phase to validate the approach and to provide a reference implementation for future specifications.
```

File: artifacts-wg/charter/charter.md [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+
+# Activities
+The working group intends to carry out (but is not limited to) the following:
+* Gather stakeholders using OCI specifications distribution-spec and image-spec to package and distribute artifacts to document their bundle layouts and content schemas and seek common patterns.
+* Gather stakeholders using non-OCI formats to document their layout and schemas and seek synergies and opportunities. Focus on these artifacts types in priority order:
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:33]:
```suggestion
* Gather stakeholders using non-OCI formats to document their layout and schemas to seek synergies and opportunities. Focus on the following artifacts types in priority order:
```

File: artifacts-wg/charter/charter.md [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+  1. Cloud-native artifact types not currently packaged in OCI, e.g. those in ArtifactHub [3]
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:34]:
```suggestion
  1. Cloud-native artifact types not currently packaged in OCI format, e.g. those in ArtifactHub [3]
```

File: artifacts-wg/charter/charter.md [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+  2. Software library artifacts such as those in npm, pyPi and Maven Central
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:35]:
```suggestion
  2. Software library artifacts, such as those in npm, PyPi, and Maven Central
```

File: artifacts-wg/charter/charter.md [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+  3. System package artifacts such as those in rpm, deb, brew
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:36]:
```suggestion
  3. System package artifacts, such as those in RPM, deb, and brew repositories
```

File: artifacts-wg/charter/charter.md [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+* Advocate for and contribute to common formats to enable search and discovery of attributes and aspects of artifacts like SBOMs, attestations and other elements
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:37]:
```suggestion
* Advocate for and contribute to common formats to enable search and discovery of attributes and aspects of artifacts like SBOM's, attestations, and other elements
```

File: artifacts-wg/charter/charter.md [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+* Establish "duck types" [4] for common attributes
+* Demonstrate via prototypes use of published schemas to facilitate query and analysis of bundles and content.
```
Comment (sabre1041) [inline artifacts-wg/charter/charter.md:39]:
```suggestion
* Demonstrate via prototypes the use of published schemas to facilitate query and analysis of bundles and content.
```

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Love the progress the WG has made here. Made a bunch of recommendations based on review

File: artifacts-wg/charter/charter.md (hunk 15 above)
Comment (afflom) [reply artifacts-wg/charter/charter.md:6]: I'm not sure if your suggestion here captures the intent of the original text.

File: artifacts-wg/charter/charter.md (hunk 19 above)
Comment (sajayantony) [reply artifacts-wg/charter/charter.md:23]: I wouldn't mind keeping it as `this CNCF TAG complements work in other projects like OCI`

**Review by AlexsJones — APPROVED** [review]

**Review by sabre1041 — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
In the spirit of "Perfect is the enemy of good", I approve to continue making iterative progress.

File: artifacts-wg/charter/charter.md (hunk 3 above)
Comment (rchincha) [reply artifacts-wg/charter/charter.md]: That,

1) all "artifacts" use OCI artifacts path and by extension use the dist-spec referrers facility

- OR -

2) all "artifacts" use https://github.com/opencontainers/artifacts (or derivative) and/or implement specs coming out of this WG

remains unresolved IMO.

I suspect this may just turn out to be a cost-benefit business decision in the end. Too costly to do 2) so just do 1) for example. But not sure we want to make this assumption so early in the game.

File: artifacts-wg/charter/charter.md (hunk 19 above)
Comment (FeynmanZhou) [reply artifacts-wg/charter/charter.md:23]: Agree with @sabre1041 and @sajayantony  . I think this WG should not be limited to specifications in OCI only, it would be flexible to mention that OCI spec is an example initiative. I guess the Artifact WG may also has its own Spec or contribute to other project specifications in the future.

**Review by justincormack — APPROVED** [review]
LGTM

File: artifacts-wg/charter/charter.md (hunk 15 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md:6]: I think there's a significant difference between "offering" and "to offer". The idea here is that the group will bring together projects that _already_ offer bundle and packaging capabilities but currently follow their own siloed patterns and practices. We intend to bring these groups closer together in spirit _and_ implementations.

I'm fine with the latter changes but I think we should leave "offering" as is.

File: artifacts-wg/charter/charter.md (hunk 8 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md:9]: @AlexsJones I believe the group intends to influence common patterns directly. First they'll gather up many existing practices, then they'll share practices and patterns across them, and finally they'll influence and standardize these common practices in OCI specifications.

File: artifacts-wg/charter/charter.md (hunk 16 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md:8]: There's a line "Advocate for innovative projects", is that sufficient @sabre1041?

I think the group intends to evangelize projects and products that contribute to evolution of and innovation in standard practices and patterns.

File: artifacts-wg/charter/charter.md (hunk 19 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md:23]: @sabre1041 @FeynmanZhou makes sense, I agree this line makes it sound like we're focusing exclusively on OCI bundles.

Perhaps we could add to the end of the sentence: 

"The work ... in this CNCF TAG complements work on specifications in ~~the~~ Open Containers Initiative (OCI) ~~itself~~, but will consider other packaging formats and not be limited only to OCI."

File: artifacts-wg/charter/appendix.md (hunk 12 above)
Comment (joshgav) [reply artifacts-wg/charter/appendix.md:7]: I'm fine with these changes or the original. @sabre1041 in meetings we agreed that we're going to iterate on these use cases as one of the first tasks of the established WG.

File: artifacts-wg/charter/charter.md (hunk 2 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md]: To be frank I expect we'll keep static registration but introduce experiments for dynamic type/schema discovery alongside that.

I think we've removed this topic from the charter in any case, LGTM.

File: artifacts-wg/charter/charter.md (hunk 3 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md]: Thanks for this discussion, it's helpful to understand the potential issues with existing artifacts and the evolution of image-spec. Thanks for participating in this group @sudo-bmitch.

File: artifacts-wg/charter/charter.md (hunk 5 above)
Comment (joshgav) [reply artifacts-wg/charter/charter.md]: FWIW there's a comparison to be made with JSON schema and <https://www.schemastore.org/json/>. VS Code and other tools rely on schemastore for type checks and coding hints; perhaps runtime tools also verify manifests against published schemas. Of course, schemastore is a static registration system like IANA. My point is there does seem to be value in enabling many tools and services to discover the types and schemas behind manifests.

**Review by joshgav — APPROVED** [review]
Really like how you refactored the Goals section @afflom. You captured the intent of the original list items but described them more clearly in human terms.

LGTM!
